docs/JAX_porting_HF_model.md

```diff
@@ -0,0 +1,453 @@
+    8e2a389dd7064d8381eded838501edf7, 862c4ada4adf422285d0f4184402b417, 9c39b737455647b7a90d069a2d5d1174,
+    84abf9c99e0a4ad18669c931ea9f1225, 3aad07fed4a64c1fbce16f65168c5942, a2e56693c27247ab8b19b75af5568aec,
+    a9d25d395b7240c699ba1a001e97bc01, 115307c0e18f4e7c9cb03b40dc045fea, 0075cf237eb1492091f3b622441acaec,
+    362c8f5b725647f1baf87a177625e914, 55c6ab804ced4da988f530c8f84a3aa4, 897a5b9a5c8a40d1b90058bb1ffea1fa,
+    0c5b0c8624b64e899bfd9dcc73ba1345, 1f8be54d549b4b6684081dcb5deded73, 6776f54ea4b24332ae9650bf651052ac,
+    b1aba6f7ab6f4d57bf678d113df239bf, e6aa29e3e18845efaf5b1c66511d6504, 3b4cc5d3389e4cfc88b3e26ee8ba8e50,
+    8723f6a787664da1ab849d04949cd764, 19b48ca524ec43359e3e68cf1ab965b6, d1c3feea5cbc48b69d15b3cd3195f6c1,
+    bb6cee0552994adfaefeab4ffafc59bc, cbe3094b322140ee9bbf1bac829ab963, 84998fc5b9014df3b43bc42c729152a0,
+    5fd802b91c28472693ead7723cdedd5f, 265d35cf0dae4d63a7b4f2a7b160098b, d2c13617a5e84498b01ca97370e9a825,
+    cc30646ad6e443318977e939cb43be25, e749c6a737bc4168a0e280335a4639cb, 5f440b28cae6429f8614a2a930867230,
+    ad00a584a1484b1aa6295a4395f4c488, 2b018b81a8a14ff8a103f7395ce65147, 7501a49606b346e997647375b1660377,
+    a6ee1a55467b4027ab526683ce1debe9, e6e24411eac94287aa7bb1ffb6e15d0d, 3edfed06e8f24ccea0ffa6386244c05a,
+    23b0ae2153764e39a737abd5c80d3d21, 6d4aa5dfa939462b9f53a56c41803c0e, 2d81309748f34ad28e0c2f7fc2734c8f,
+    3a75b8e419004b36a06e556fa5a4e535, 48d5fa17a0c3444795708ac8cb5e7f45, b21b452dbdbe4db1a6a8c7ad75553b1b,
+    11e528bfe36341adbc264a6ca05d09d9, ddc376976d7f41b6a5dab76c32f62d57, ed4dc67131144c228f088202197e7015,
+    33ae1220d30d46c19893ed23c230995b, 2dd3319a37fe426daed81f7f3ebba034, ac443b375fe448c3bf028755064e0cc6,
+    98454d975f5842eeb4a4e11173e272fd, 990cb01cfa0c44b4966c45358e516076]
+id: oVAJOEEOzEA9
+outputId: 5df4bc65-6031-44a8-9233-6b9e1ea1d069
+---
+model_id = "meta-llama/Meta-Llama-3-8B"
+path_to_model_weights = os.path.join("/content", model_id)
+snapshot_download(repo_id=model_id, local_dir=path_to_model_weights)
+```
+
++++ {"id": "4Xo6QyVR0UkF"}
+
+Then extract the model weights from the safetensors file and store them in the `weights` dict. These weights will be loaded into our JAX model soon.
+
+```{code-cell}
+:id: pERpzsPS0fLj
+
+def load_safetensors():
+    weights = {}
+    safetensors_files = Path(path_to_model_weights).glob('*.safetensors')
+
+    for file in safetensors_files:
+        with safe_open(file, framework="jax", device="cpu") as f:
+            for key in f.keys():
+                weights[key] = f.get_tensor(key)
+    return weights
+
+weights = load_safetensors()
+```
+
++++ {"id": "6ToKOCWIPJl1"}
+
+Note that the weights are stored in bfloat16.
+
++++ {"id": "vQ7Bmvj70jjA"}
+
+## Define the model
+
++++ {"id": "HdHQwfSO1xfY"}
+
+Now we can define the model. This [diagram](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/_images/transformer_vs_llama.svg) from NVidia visualizes the model architecture pretty nicely.
+
```

Comment:
Not sure if this is a helpful reference, please feel free to include/exclude as you see fit.:)

```suggestion

You can check out the [Flax model implementation in the Transformer's library](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_flax_llama.py) for reference.

```

Reply:
This is already mentioned in the section 'load the model weights' above.